shell.nix

```diff
@@ -0,0 +1,65 @@
+{ nixroot  ? (import <nixpkgs> {}) }:
+let inherit (nixroot) stdenv
+    chromaprint fftw flac libid3tag libmad libopus libshout libsndfile
+    libusb1 libvorbis libebur128 pkgconfig portaudio portmidi protobuf qt4 qt5
+    rubberband scons sqlite taglib soundtouch vamp opusfile hidapi upower ccache git;
+  
+in stdenv.mkDerivation rec {
+  name = "mixxx-${version}";
+  version = "2.1.0-dev";
+
+  shellHook =  ''
+    export out=lin_build
+    mkdir -p $out
+    export CC="ccache $CC"
+    export CXX="ccache $CXX"
+    export out="local_build"
+
+    build() {
+      scons $sconsFlags prefix=~/mixxx $@
+    }
+
+    run() {
+      ./lin*_build/mixxx --settingsPath ./devsettings/ --resourcePath ./res $@
```

Comment:
I'm affraid this references two files if you previously build 32bits AND 64bits versions.
having a way to pass the bitwidth as argument would be nice

Reply:
I so don't like this solution, but have not come up with something better. maybe scons creats a softlink to the last build target in a common place.